go/src/koding/config/genconfig.go

```diff
@@ -0,0 +1,233 @@
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+var (
+	input  = flag.String("i", "-", "")
+	output = flag.String("o", "-", "")
+)
+
+var t = template.Must(template.New("").Parse(`// This file is auto-generated. DO NOT EDIT!
```

Comment:
>Yes, I'd probably use the solution you provided if I was creating new workers. This is completely fine.

If you could use that for new workers, you can use for the old ones as well.

Yep you are saying we have a different use case but i dont see it as different. I see it as just complex.

The complexity you create comes from the first design decision that was made by trying to use "strings" to get the configuration you are looking for. And to remediate that complexity you are adding another level of complexity by creating templates on the fly and adding them as "generated" files. Another level of complexity you add comes with "code generation" part, not to use json serialization you have implemented 477 more lines of code ( i dont count the lines usually, just stating here because previously you mentioned about LOC )

What i wanted to see with my request was simple config generation on configure step, i am not against using bindata.

>  I see no advantages in such solution.

You will have a completely build-time-safe structs and you can refactor them easily whenever you want. We had great success with using multiconfig on many places.

For the future generations lets have simple and understandable codebase.